### PR TITLE
Fixed jvm thread attach leak, updated dependencies

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-common-runtime/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ set(AWS_DEPS_BUILD_DIR "${CMAKE_BINARY_DIR}/build" CACHE PATH "Dependencies buil
 set(AWS_DEPS_DOWNLOAD_DIR "${AWS_DEPS_BUILD_DIR}/downloads" CACHE PATH "Dependencies download directory.")
 
 set(AWS_C_COMMON_URL "https://github.com/awslabs/aws-c-common.git")
-set(AWS_C_COMMON_SHA "v0.4.5")
+set(AWS_C_COMMON_SHA "4031ef190ae8c0141c0e652279eb511ceb4afece")
```
Comment (JonathanHenson) [inline aws-common-runtime/CMakeLists.txt]: why is this not a tag?

File: src/main/java/software/amazon/awssdk/crt/http/CrtHttpStreamHandler.java [hunk 2]
```diff
@@ -41,7 +41,7 @@
      * @param stream The HttpStream object
      * @param hasBody True if the HTTP Response had a Body, false otherwise.
```
Comment (JonathanHenson) [inline src/main/java/software/amazon/awssdk/crt/http/CrtHttpStreamHandler.java]: comment looks out of date.

File: src/native/crt.c [hunk 3]
```diff
@@ -202,7 +203,20 @@ struct aws_string *aws_jni_new_string_from_jstring(JNIEnv *env, jstring str) {
...
+}
+
+void s_attach_jvm_cleanup_to_thread(void *user_data) {
+    AWS_FATAL_ASSERT(AWS_OP_SUCCESS == aws_thread_current_at_exit(s_detach_jvm_from_thread, user_data));
+}
+
+static aws_thread_once s_jvm_once = AWS_THREAD_ONCE_STATIC_INIT;
+
 JNIEnv *aws_jni_get_thread_env(JavaVM *jvm) {
+    aws_thread_call_once(&s_jvm_once, s_attach_jvm_cleanup_to_thread, jvm);
```
Comment (JonathanHenson) [inline src/native/crt.c]: I'm pretty sure I wrote this to emulate std::call_once, which means it gets called once per process, not once per thread. Is that intended?

File: aws-common-runtime/CMakeLists.txt (hunk 1 above)
Comment (justinboswell) [reply aws-common-runtime/CMakeLists.txt]: Waiting on the PR

File: src/native/crt.c (hunk 3 above)
Comment (justinboswell) [reply src/native/crt.c]: Nope, I was so annoyed at LTO I forgot how to computer, will fix

File: src/main/java/software/amazon/awssdk/crt/http/CrtHttpStreamHandler.java (hunk 2 above)
Comment (justinboswell) [reply src/main/java/software/amazon/awssdk/crt/http/CrtHttpStreamHandler.java]: Bah! Will fix.

**Pull request merged**